### PR TITLE
feat(chore): Adding the get_platform helper

### PR DESCRIPTION
**Pull request by edgarsandi**
feat(chore) add get_platform helper

Related #1089

### CODE REVIEW

File: config/config_helpers_test.go [hunk 1]
```diff
@@ -480,6 +481,16 @@ func toStringSlice(t *testing.T, value interface{}) []string {
...
 
+func TestGetPlatform(t *testing.T) {
+	t.Parallel()
+
+	expectedPlatform := runtime.GOOS
+	actualPlatform, err := getPlatform(nil, nil)
+
+	assert.Nil(t, err)
+	assert.Equal(t, expectedPlatform, actualPlatform)
+}
```
Comment (brikis98) [inline config/config_helpers_test.go:492]: This unit test doesn't do much, as it's almost a copy/paste of the actual function 😁 

An integration test that uses this helper in a `.hcl` file would be a bit better of a test.

**Comment by edgarsandi** [comment]
> Thanks for the PR! Could you update the docs too?

Sure! I'll update too!

File: config/config_helpers_test.go (hunk 1 above)
Comment (edgarsandi) [reply config/config_helpers_test.go:492]: You're right!
I'll do better tests!
Thanks for your review and comments!

File: docs/_docs/04_reference/built-in-functions.md [hunk 2]
```diff
@@ -232,6 +234,31 @@ remote_state {
 
 Note that [Terraform will read environment variables](https://www.terraform.io/docs/configuration/environment-variables.html#tf_var_name) that start with the prefix `TF_VAR_`, so one way to share a variable named `foo` between Terraform and Terragrunt is to set its value as the environment variable `TF_VAR_foo` and to read that value in using this `get_env()` built-in function.
 
+## get\_platform
+
+`get_platform()` returns the current Operational System. Example:
```
Comment (brikis98) [inline docs/_docs/04_reference/built-in-functions.md:239]:
```suggestion
`get_platform()` returns the current Operating System. Example:
```

File: docs/_docs/04_reference/built-in-functions.md [hunk 3]
```diff
@@ -232,6 +234,31 @@ remote_state {
...
+``` hcl
+inputs = {
+  platform = get_platform()
+}
+```
+
+This function can also be used in a comparison to evaluate what to do based on the current operating system. Example:
+``` hcl
+output "platform" {
+  value = var.platform == "darwin" ? "" : ""
```
Comment (brikis98) [inline docs/_docs/04_reference/built-in-functions.md:250]:
```suggestion
  value = var.platform == "darwin" ? "(value for MacOS)" : "(value for other OS's)"
```

File: test/integration_test.go [hunk 4]
```diff
@@ -2401,6 +2403,30 @@ func TestAWSGetCallerIdentityFunctions(t *testing.T) {
...
+
+	require.NoError(
+		t,
+		runTerragruntCommand(t, fmt.Sprintf("terragrunt output -no-color -json --terragrunt-non-interactive --terragrunt-working-dir %s", rootPath), &stdout, &stderr),
+	)
+
+	outputs := map[string]TerraformOutput{}
+
+	require.NoError(t, json.Unmarshal([]byte(stdout.String()), &outputs))
+	assert.Equal(t, outputs["platform"].Value, runtime.GOOS)
```
Comment (brikis98) [inline test/integration_test.go:2427]: Much better test, thanks 👍 

NIT: To ensure we don't get a panic if this test uncovers a bug, probably safer to do:

```suggestion
	platform, hasPlatform := outputs["platform"]
	require.True(t, hasPlatform)
	require.Equal(t, platform.Value, runtime.GOOS)
```

File: docs/_docs/04_reference/built-in-functions.md (hunk 2 above)
Comment (edgarsandi) [reply docs/_docs/04_reference/built-in-functions.md]: fixed!

File: docs/_docs/04_reference/built-in-functions.md (hunk 3 above)
Comment (edgarsandi) [reply docs/_docs/04_reference/built-in-functions.md]: yep, It's much better!

I was in doubt to decide how to do a good doc to cover this point.

File: test/integration_test.go (hunk 4 above)
Comment (edgarsandi) [reply test/integration_test.go]: TIL: what NIT means! 😁

**Comment by brikis98** [comment]
Tests passed! Merging now.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.23.21

**Pull request merged**